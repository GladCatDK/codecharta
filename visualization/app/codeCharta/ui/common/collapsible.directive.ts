import {IDirective} from "angular";

/**
 * Container for {CollapsibleElementDirective}. Inner elements do not replace this. They are put inside the ng-transclude tag.
 */
export class CollapsibleDirective implements IDirective{

    restrict = "E";
    scope = {};
    replace = true;
    transclude = true;
    template = "<ul class='collapsible' data-collapsible='accordion' ng-transclude></ul>";

    constructor() {}

}
