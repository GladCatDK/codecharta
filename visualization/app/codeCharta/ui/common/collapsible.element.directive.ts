import {IDirective} from "angular";

/**
 * Collapsible element. Should be put inside a {@link CollapsibleDirective}.
 */
export class CollapsibleElementDirective implements IDirective{

    restrict = "E";
     scope = {
        iconClass: "@iconClass",
        label:"@"
    };

    replace = true;
    transclude = true;
    template = "<li>\n" +
        "    <div class=\"collapsible-header\"><i class=\"fa {{::iconClass}}\"></i>{{::label}}</div>\n" +
        "    <div class=\"collapsible-body\">\n" +
        "        <div class=\"row\" ng-transclude>\n" +
        "          <!--transcluded content-->\n" +
        "        </div>\n" +
        "    </div>\n" +
        "</li>";

    constructor(private $rootScope, private $timeout) {}

    /**
     * calls the materialize function collapsible(...) on the collapsible element. This is needed to initialize the functionality.
     */
    link() {
        //needs to be done
        const ctx = this;
        ($("#settingsPanel .collapsible") as any).collapsible({
            accordion: false,
            onOpen: function() { ctx.$timeout(function () {
                //This forces to rerender all rzsliders in this collapsible element. rz sliders do not automatically recognize collapse events from materialize therefore this is nessecary.
                ctx.$rootScope.$broadcast("rzSliderForceRender");
            }); }
        });
    }

}