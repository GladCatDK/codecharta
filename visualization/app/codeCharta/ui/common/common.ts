import angular from "angular";
import "./dropdown/dropdown.js";
import "./checkbox/checkbox.ts";
import "./numberField/numberField.js";
import "./collapsible/collapsible.js";
import "./collapsibleElement/collapsibleElement.js";
import "./slider/slider.js";

import {fabComponent} from "./fab.component.ts";

angular.module("app.codeCharta.ui.common", ["app.codeCharta.ui.common.dropdown", "app.codeCharta.ui.common.checkbox", "app.codeCharta.ui.common.numberField", "app.codeCharta.ui.common.collapsible", "app.codeCharta.ui.common.collapsibleElement", "app.codeCharta.ui.common.slider"])
    .component(fabComponent.selector, fabComponent);
