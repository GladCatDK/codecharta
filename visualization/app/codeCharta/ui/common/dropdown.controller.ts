"use strict";
import {TooltipService, TooltipServiceSubscriber, Tooltips} from "../../core/tooltip/tooltip.service.ts";
import {IScope} from "angular";
/**
 * Controller for the DropdownDirective
 */
class DropdownController implements TooltipServiceSubscriber{

    /* @ngInject */
    /**
     *
     * @param {TooltipService} tooltipService
     * @param {Scope} $scope
     */
    constructor(
        private tooltipService: TooltipService,
        private $scope: IScope) {

        tooltipService.subscribe(this);

    }

    onTooltipsChanged(tooltips: Tooltips, event: Event) {
        this.$scope.$apply();
    }

    /**
     * returns the tooltip description related to the given key
     * @param {String} key
     * @returns {String} description
     */
    getTooltipTextByKey(key) {
        return this.tooltipService.getTooltipTextByKey(key);
    }

}

export {DropdownController};


