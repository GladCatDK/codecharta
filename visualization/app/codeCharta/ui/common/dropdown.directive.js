"use strict";
import {DropdownController} from "./dropdown.controller.ts";

/**
 * Renders a dropdown menu.
 */
class DropdownDirective{

    /**
     * @constructor
     */
    constructor() {

        /** @type {string} */
        this.template = "<div class=\"input-field col s12\">\n" +
            "    <label for=\"dropdown-directive-{{::$id}}\">{{::ctrl.label}}</label>\n" +
            "    <br />\n" +
            "    <select class=\"browser-default\" ng-model=\"ctrl.model\" id=\"dropdown-directive-{{::$id}}\">\n" +
            "        <option title=\"{{ctrl.getTooltipTextByKey(x);}}\" ng-repeat=\"x in ctrl.values\" value=\"{{x}}\">{{x}}</option>\n" +
            "    </select>\n" +
            "</div>";

        /** @type {string} */
        this.restrict = "E";

        /**
         * Binds 'label' string. Binds 'values' array with all possible values. Binds 'model' with the current selected value. Binds a 'change' function.
         * @type {Scope}
         **/
        this.scope = {
            label: "@",
            values: "=",
            model: "=",
            change: "&"
        };

        /**
         *
         * @type {DropdownController}
         */
        this.controller = DropdownController;

        /**
         *
         * @type {string}
         */
        this.controllerAs = "ctrl";

        /**
         *
         * @type {boolean}
         */
        this.bindToController = true;

    }

    /**
     * Links the change function to model changes.
     * @param {Scope} scope
     */
    link(scope){
        scope.$watch(
            ()=>{return scope.ctrl.model;},
            ()=>{scope.ctrl.change();}
        );
    }

}

export {DropdownDirective};
