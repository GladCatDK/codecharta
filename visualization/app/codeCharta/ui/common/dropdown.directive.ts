import {DropdownController} from "./dropdown.controller.ts";
import {IDirective} from "angular";

/**
 * Renders a dropdown menu.
 */
export class DropdownDirective implements IDirective{

    template = "<div class=\"input-field col s12\">\n" +
        "    <label for=\"dropdown-directive-{{::$id}}\">{{::ctrl.label}}</label>\n" +
        "    <br />\n" +
        "    <select class=\"browser-default\" ng-model=\"ctrl.model\" id=\"dropdown-directive-{{::$id}}\">\n" +
        "        <option title=\"{{ctrl.getTooltipTextByKey(x);}}\" ng-repeat=\"x in ctrl.values\" value=\"{{x}}\">{{x}}</option>\n" +
        "    </select>\n" +
        "</div>";

    restrict = "E";

    scope = {
        label: "@",
        values: "=",
        model: "=",
        change: "&"
    };


    controller = DropdownController;
    controllerAs = "ctrl";
    bindToController = true;


    constructor() {}

    /**
     * Links the change function to model changes.
     * @param {Scope} scope
     */
    link(scope){
        scope.$watch(
            ()=>{return scope.ctrl.model;},
            ()=>{scope.ctrl.change();}
        );
    }

}