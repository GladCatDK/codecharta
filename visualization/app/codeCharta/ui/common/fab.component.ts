import $ from "jquery";

/**
 * Renders a simple floating action button.
 */
export const fabComponent = {
    selector: "fabComponent",
    template: "<a class='btn-floating btn-large waves-effect waves-light teal'><i class='fa {{::$ctrl.iconClass}}'></i></a>",
    bindings: {
        iconClass: "@"
    }
};


