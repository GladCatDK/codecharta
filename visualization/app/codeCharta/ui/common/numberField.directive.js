"use strict";

/**
 * Renders a number field.
 */
class NumberFieldDirective {

    constructor() {
        /** @type {string} **/
        this.template = "<div class=\"input-field col s12\">\n" +
            "    <label for=\"numberField-directive-{{::$id}}\">{{::label}}</label>\n" +
            "    <br />\n" +
            "    <input id=\"numberField-directive-{{::$id}}\" type=\"number\" ng-model=\"model\" min={{::min}}>\n" +
            "</div>";

        /** @type {string} **/
        this.restrict = "E";

        /**
         * Binds 'model', 'label', a minimal value 'min' and a 'change' function.
         * @type {Scope}
         **/
        this.scope = {
            model: "=",
            label: "@",
            min: "@",
            change:"&"
        };
    }

    /**
     * Links the change function to model changes.
     * @param {Scope} scope
     */
    link(scope){
        scope.$watch(
            ()=>{return scope.model;},
            ()=>{scope.change();}
        );
    }
    
}

export {NumberFieldDirective};

