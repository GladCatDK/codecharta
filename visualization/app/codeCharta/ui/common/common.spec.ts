import {IRootScopeService, ICompileService, IScope} from "angular";

import "./common";
import {NG, NGMock} from "../../../../mocks/ng.mockhelper";

describe("common", ()=>{

    let $rootScope: IRootScopeService;
    let $compile: ICompileService;


    beforeEach(NGMock.mock.module("app.codeCharta.ui.common"));

    beforeEach(NGMock.mock.inject((_$rootScope_, _$compile_) => {
        $rootScope = _$rootScope_;
        $compile = _$compile_;
    }));

    describe("checkbox", ()=>{

        it("snapshot test", () => {
            let scope: IScope = $rootScope.$new();
            let element = $compile(NG.element('<checkbox-directive label="some label" model="model" change="change()"></checkbox-directive>'))(scope);
            scope.$apply();
            expect(element).toMatchSnapshot();
        });

    });

    describe("collapsible", ()=>{

        it("snapshot test", () => {
            let scope: IScope = $rootScope.$new();
            let element = $compile(NG.element('<collapsible-directive>SOMETHING</collapsible-directive>'))(scope);
            scope.$apply();
            expect(element).toMatchSnapshot();
        });

    });

    describe("dropdown", ()=>{

        it("snapshot test", () => {
            let scope: IScope = $rootScope.$new();
            let element = $compile(NG.element('<dropdown-directive label="label" value="value" model="value"></dropdown-directive>'))(scope);
            scope.$apply();
            expect(element).toMatchSnapshot();
        });

    });

    describe("numberField", ()=>{

        it("snapshot test", () => {
            let scope: IScope = $rootScope.$new();
            let element = $compile(NG.element('<number-field-directive label="label" value="10" model="value" min="2"></number-field-directive>'))(scope);
            scope.$apply();
            expect(element).toMatchSnapshot();
        });

    });

    describe("slider", ()=>{

        it("snapshot test", () => {
            let scope: IScope = $rootScope.$new();
            let element = $compile(NG.element('<slider-directive label="label" value="10" model="value" min="2" max="4" step="1" decimal="0"></slider-directive>'))(scope);
            scope.$apply();
            expect(element).toMatchSnapshot();
        });

    });

    describe("collapsibleElement", ()=>{

        //TODO materialize again ...
        xit("snapshot test", () => {
            let scope: IScope = $rootScope.$new();
            let element = $compile(NG.element('<collapsible-directive><collapsible-element-directive label="some" icon-class="class">SOMETHING</collapsible-element-directive></collapsible-directive>'))(scope);
            scope.$apply();
            expect(element).toMatchSnapshot();
        });

    });

});


