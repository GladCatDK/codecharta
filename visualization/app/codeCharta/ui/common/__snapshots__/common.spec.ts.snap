// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common checkbox snapshot test 1`] = `
Object {
  "0": <checkbox-directive
    label="some label"
    model="model"
    change="change()"
    class="ng-scope ng-isolate-scope"
>
    <div
        class="input-field col s12"
    >
        <input
            type="checkbox"
            ng-model="model"
            id="checkbox-directive-3"
            class="ng-pristine ng-untouched ng-valid ng-empty"
        />
        <label
            for="checkbox-directive-3"
            class="ng-binding"
        >
            some label
        </label>
    </div>
</checkbox-directive>,
  "length": 1,
}
`;

exports[`common collapsible snapshot test 1`] = `
Object {
  "0": <ul
    class="collapsible ng-scope ng-isolate-scope"
    data-collapsible="accordion"
    ng-transclude=""
>
    SOMETHING
</ul>,
  "length": 1,
}
`;

exports[`common dropdown snapshot test 1`] = `
Object {
  "0": <dropdown-directive
    label="label"
    value="value"
    model="value"
    class="ng-scope ng-isolate-scope"
>
    <div
        class="input-field col s12"
    >
        <label
            for="dropdown-directive-13"
            class="ng-binding"
        >
            label
        </label>
        <br />
        <select
            class="browser-default ng-pristine ng-untouched ng-valid ng-empty"
            ng-model="ctrl.model"
            id="dropdown-directive-13"
        >
            <option
                value="? undefined:undefined ?"
                selected="selected"
            />
            <!-- ngRepeat: x in ctrl.values -->
        </select>
    </div>
</dropdown-directive>,
  "length": 1,
}
`;

exports[`common fab snapshot test 1`] = `
Object {
  "0": <fab-component
    icon-class="some class"
    class="ng-scope ng-isolate-scope"
>
    <a
        class="btn-floating btn-large waves-effect waves-light teal"
    >
        <i
            class="fa some class"
        />
    </a>
</fab-component>,
  "length": 1,
}
`;

exports[`common numberField snapshot test 1`] = `
Object {
  "0": <number-field-directive
    label="label"
    value="10"
    model="value"
    min="2"
    class="ng-scope ng-isolate-scope"
>
    <div
        class="input-field col s12"
    >
        <label
            for="numberField-directive-16"
            class="ng-binding"
        >
            label
        </label>
        <br />
        <input
            id="numberField-directive-16"
            type="number"
            ng-model="model"
            min="2"
            class="ng-pristine ng-untouched ng-valid ng-empty ng-valid-min"
        />
    </div>
</number-field-directive>,
  "length": 1,
}
`;

exports[`common slider snapshot test 1`] = `
Object {
  "0": <slider-directive
    label="label"
    value="10"
    model="value"
    min="2"
    max="4"
    step="1"
    decimal="0"
    class="ng-scope ng-isolate-scope"
>
    <div
        class="range-field col s12"
    >
        <label
            for="slider-directive-19"
            class="ng-binding"
        >
            label
        </label>
        <br />
        <input
            id="slider-directive-19"
            type="range"
            ng-model="model"
            min="2"
            max="4"
            step="1"
            class="ng-pristine ng-untouched ng-valid ng-empty ng-valid-min ng-valid-max ng-valid-step"
        />
    </div>
</slider-directive>,
  "length": 1,
}
`;
